WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift

```diff
@@ -401,16 +401,24 @@ private extension StoreStatsAndTopPerformersViewController {
 }
 
 private extension StoreStatsAndTopPerformersViewController {
-    func showSiteVisitors(_ shouldShowSiteVisitors: Bool) {
+    func updateSiteVisitors(mode: SiteVisitStatsMode) {
         periodVCs.forEach { vc in
-            vc.shouldShowSiteVisitStats = shouldShowSiteVisitors
+            vc.siteVisitStatsMode = mode
         }
     }
 
     func handleSiteVisitStatsStoreError(error: SiteVisitStatsStoreError) {
         switch error {
-        case .statsModuleDisabled, .noPermission:
-            showSiteVisitors(false)
+        case .noPermission:
+            updateSiteVisitors(mode: .hidden)
+        case .statsModuleDisabled:
+            let defaultSite = ServiceLocator.stores.sessionManager.defaultSite
```

Comment:
Thanks for noticing the risk! I think the best and simplest solution would be to force reload stats on change of site - since only after reloading can we get the error for fetching visitor stats. I have added the fix in 3d92789c3b462f9fa91fb6eaf7cb270982c7f3a2, let me know what you think.

Reply:
this looks like a much simpler solution! 👍